### ISSUE TITLE
gorm doesnt throw error when ambiguous selector, inconsistent results.

### ISSUE THREAD

**Issue by robert-king** [jinzhu/gorm#2917]
when saving with .Find() to a struct that has multiple embedded structs with the same fields with queries that join tables with the same fields, GORM inconsistently writes the wrong value to the wrong struct.

```
type NarratorInfo struct {
	model.Narrator `json:"narrator"`
	model.Person `json:"person"`
	model.PersonTranslation `json:"personTranslation"`
}
```

Note that narratorInfo.ID would be an "inconsistent selector" because It doesn't know if it's narratorInfo.Narrator.ID or narratorInfo.Person.ID or is it narratorInfo.PersonTranslation.ID
I thought gorm could handle this by knowing the table name.. e.g. select narrator.*, person.*,.. it could write the values from narrator.* to Narrator and so on.
It worked at first, but then changing the order of the struct properties breaks it.

The full query generated by gorm:
```
SELECT narrator.*, person.*, person_translation.*
FROM `narrator`
         inner join person on narrator.person_id = person.id
         left join person_translation on person.id = person_translation.person_id
WHERE `narrator`.`deleted_at` IS NULL
  AND ((person_translation.language_id = 1) AND (narrator.track_id = 5))
```

and if i run the query I get the two following ids:
```
person.id = person_translation.person_id = narrator.person_.id = 2
person.id = person_translation.person_id = narrator.person_.id = 3
```

so after GORM decodes
```
var narrators []NarratorInfo
using Find(&narrators)
```

sometimes I get narrators[0].person.id = 1, which should only ever be 2 or 3. This is because it's saving narrator.id to person.id and person.id to narrator.id, this depends on the order of the columns or struct fields

There are one of two things gorm should do:
Either, Gorm should throw an error saying it doesn't know what to do, since Person and Narrator both have an ID, it should say ambiguous selector 

Otherwise, Gorm should know which struct to write to based on the column name, e.g. narrator.* goes to Narrator and so on. It should be documented that this isn't a feature of GORM to write to multiple embedded structs, as it's possible to do it with Xorm. It seems I'll have to manually type out every single field value every time.

```
type Person struct {
	ID          int       `gorm:"column:id;primary_key" json:"id"`
	CreatedAt   time.Time `gorm:"column:created_at" json:"-"`
	UpdatedAt   time.Time `gorm:"column:updated_at" json:"-"`
	DeletedAt   null.Time `gorm:"column:deleted_at" json:"-"`
	CreatedBy   int       `gorm:"column:created_by" json:"-"`
	UpdatedBy   int       `gorm:"column:updated_by" json:"-"`
	ImageFileID null.Int  `gorm:"column:image_file_id" json:"imageFileId"`
	Gender      int       `gorm:"column:gender" json:"gender"`
}

// TableName sets the insert table name for this struct type
func (p *Person) TableName() string {
	return "person"
}


type PersonTranslation struct {
	ID          int         `gorm:"column:id;primary_key" json:"id"`
	CreatedAt   time.Time   `gorm:"column:created_at" json:"-"`
	UpdatedAt   time.Time   `gorm:"column:updated_at" json:"-"`
	DeletedAt   null.Time   `gorm:"column:deleted_at" json:"-"`
	CreatedBy   int         `gorm:"column:created_by" json:"-"`
	UpdatedBy   int         `gorm:"column:updated_by" json:"-"`
	PersonID    int         `gorm:"column:person_id" json:"personId"`
	LanguageID  int         `gorm:"column:language_id" json:"languageId"`
	FirstName   null.String `gorm:"column:first_name" json:"firstName"`
	LastName    null.String `gorm:"column:last_name" json:"lastName"`
	DisplayName null.String `gorm:"column:display_name" json:"displayName"`
	Description null.String `gorm:"column:description" json:"description"`
}

// TableName sets the insert table name for this struct type
func (p *PersonTranslation) TableName() string {
	return "person_translation"
}


type Narrator struct {
	ID              int       `gorm:"column:id;primary_key" json:"id"`
	CreatedAt       time.Time `gorm:"column:created_at" json:"-"`
	UpdatedAt       time.Time `gorm:"column:updated_at" json:"-"`
	DeletedAt       null.Time `gorm:"column:deleted_at" json:"-"`
	CreatedBy       int       `gorm:"column:created_by" json:"-"`
	UpdatedBy       int       `gorm:"column:updated_by" json:"-"`
	StatusCode      int       `gorm:"column:status_code" json:"statusCode"`
	ReleaseDatetime null.Time `gorm:"column:release_datetime" json:"releaseDatetime"`
	TrackID         int       `gorm:"column:track_id" json:"trackId"`
	PersonID        int       `gorm:"column:person_id" json:"personId"`
}

// TableName sets the insert table name for this struct type
func (n *Narrator) TableName() string {
	return "narrator"
}

```

**Comment by robert-king** [comment]
these two aren't the same, but I can see this is probably a 2.0 milestone, so maybe this can be fixed in 2.0?

https://github.com/jinzhu/gorm/pull/2768
https://github.com/jinzhu/gorm/issues/1504

**Comment by robert-king** [comment]
fyi i'm trying to solve this with "embedded" or "embedded_prefix" although there isn't any documentation 
https://github.com/jinzhu/gorm/blob/master/embedded_struct_test.go
